Tools/SandboxTest.ps1

```diff
@@ -1,269 +1,518 @@
+$script:UiLibsDownloadUrl_v2_7 = 'https://github.com/microsoft/microsoft-ui-xaml/releases/download/v2.7.3/Microsoft.UI.Xaml.2.7.x64.appx'
+$script:UiLibsDownloadUrl_v2_8 = 'https://github.com/microsoft/microsoft-ui-xaml/releases/download/v2.8.6/Microsoft.UI.Xaml.2.8.x64.appx'
+$script:UiLibsDownloadUrl_NuGet = 'https://globalcdn.nuget.org/packages/microsoft.ui.xaml.2.8.6.nupkg?packageVersion=2.8.6'
+
+# Expected Hashes
+$script:VcLibsHash = 'B56A9101F706F9D95F815F5B7FA6EFBAC972E86573D378B96A07CFF5540C5961'
+$script:UiLibsHash_v2_7 = '8CE30D92ABEC6522BEB2544E7B716983F5CBA50751B580D89A36048BF4D90316'
+$script:UiLibsHash_v2_8 = '249D2AFB41CC009494841372BD6DD2DF46F87386D535DDF8D9F32C97226D2E46'
+$script:UiLibsHash_NuGet = '6B62BD3C277F55518C3738121B77585AC5E171C154936EC58D87268BBAE91736'
+
+# File Paths
+$script:AppInstallerDataFolder = Join-Path -Path $env:LOCALAPPDATA -ChildPath 'Packages' -AdditionalChildPath $script:AppInstallerPFN
+$script:DependenciesCacheFolder = Join-Path -Path $script:AppInstallerDataFolder -ChildPath "$script:ScriptName.Dependencies"
+$script:TestDataFolder = Join-Path -Path $script:AppInstallerDataFolder -ChildPath $script:ScriptName
+$script:PrimaryMappedFolder = (Resolve-Path -Path $MapFolder).Path
+$script:ConfigurationFile = Join-Path -Path $script:TestDataFolder -ChildPath "$script:ScriptName.wsb"
+
+# Sandbox Settings
+$script:SandboxDesktopFolder = 'C:\Users\WDAGUtilityAccount\Desktop'
+$script:SandboxWorkingDirectory = Join-Path -Path $script:SandboxDesktopFolder -ChildPath $($script:PrimaryMappedFolder | Split-Path -Leaf)
+$script:SandboxTestDataFolder = Join-Path -Path $script:SandboxDesktopFolder -ChildPath $($script:TestDataFolder | Split-Path -Leaf)
+$script:SandboxBootstrapFile = Join-Path -Path $script:SandboxTestDataFolder -ChildPath "$script:ScriptName.ps1"
+$script:HostGeoID = (Get-WinHomeLocation).GeoID
+
+# Misc
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+$script:WebClient = New-Object System.Net.WebClient
+$script:CleanupPaths = @()
+
+# The experimental features get updated later based on a switch that is set
+$script:SandboxWinGetSettings = @{
+    '$schema'            = 'https://aka.ms/winget-settings.schema.json'
+    logging              = @{
+        level = 'verbose'
+    }
+    experimentalFeatures = @{
+        fonts = $false
+    }
+}
 
-if (-Not (Test-Path -Path $mapFolder -PathType Container)) {
-  Write-Error -Category InvalidArgument -Message 'The provided MapFolder is not a folder.'
+####
+# Description: Ensures that a folder is present. Creates it if it does not exist
+# Inputs: Path to folder
+# Outputs: Boolean. True if path exists or was created; False if otherwise
+####
+function Invoke-CleanExit {
+    param (
+        [Parameter(Mandatory = $true)]
+        [int] $ExitCode
+    )
+    Invoke-FileCleanup -FilePaths $script:CleanupPaths
+    $script:WebClient.Dispose()
+    Write-Debug "Exiting ($ExitCode)"
+    exit $ExitCode
 }
```

Comment:
The description of this function doesn't seem right. It is of `Initialize-Folder`.